shuup/core/models/_orders.py

```diff
@@ -1163,5 +1278,33 @@ def get_available_payment_methods(self):
             m for m in PaymentMethod.objects.available(shop=self.shop, products=product_ids) if m.is_available_for(self)
         ]
 
+    def change_status(self, next_status: OrderStatus, user: User = None, description: str = None):
+        # validate next_status is valid or not
+
+        # if changing to the same status, then return immediately
+        if next_status == self.status:
+            return
+
+        if not self.status.allowed_next_statuses.filter(identifier=next_status.identifier).exists():
+            raise InvalidOrderStatusError(_("Error! Can not change to this status"))
+
+        # Have to do this as we will assign new value to current status
+        old_status = OrderStatus.objects.filter(identifier=self.status.identifier).first()
```

Comment:
I don't think this should be needed. At least there should be some better way than querying the database again for information we already have

Reply:
`old_status = self.status`, is the same, right? is a db query needed?

Reply:
I agree